flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java

```diff
@@ -0,0 +1,214 @@
+
+	// Both Flink's Gauge and Meter values are taken as gauge in Datadog
+	private final Map<Gauge, DGauge> gauges = new ConcurrentHashMap<>();
+	private final Map<Counter, DCounter> counters = new ConcurrentHashMap<>();
+	private final Map<Meter, DMeter> meters = new ConcurrentHashMap<>();
+
+	private DatadogHttpClient client;
+	private List<String> configTags;
+
+	public static final String API_KEY = "apikey";
+	public static final String TAGS = "tags";
+
+	@Override
+	public void notifyOfAddedMetric(Metric metric, String metricName, MetricGroup group) {
+		final String name = group.getMetricIdentifier(metricName, this);
+
+		List<String> tags = new ArrayList<>(configTags);
+		tags.addAll(getTagsFromMetricGroup(group));
+
+		synchronized (this) {
+			if (metric instanceof Counter) {
+				Counter c = (Counter) metric;
+				counters.put(c, new DCounter(c, name, tags));
+			} else if (metric instanceof Gauge) {
+				Gauge g = (Gauge) metric;
+				gauges.put(g, new DGauge(g, name, tags));
+			} else if(metric instanceof Meter) {
+				Meter m = (Meter) metric;
+				// Only consider rate
+				meters.put(m, new DMeter(m, name, tags));
+			} else if (metric instanceof Histogram) {
+				LOGGER.warn("Cannot add {} because Datadog HTTP API doesn't support Histogram", metricName);
+			} else {
+				LOGGER.warn("Cannot add unknown metric type {}. This indicates that the reporter " +
+					"does not support this metric type.", metric.getClass().getName());
+			}
+		}
+	}
+
+	@Override
+	public void notifyOfRemovedMetric(Metric metric, String metricName, MetricGroup group) {
+		synchronized (this) {
+			if (metric instanceof Counter) {
+				counters.remove(metric);
+			} else if (metric instanceof Gauge) {
+				gauges.remove(metric);
+			} else if (metric instanceof Meter) {
+				meters.remove(metric);
+			} else if (metric instanceof Histogram) {
+				// No Histogram and Meter metrics are registered
+			} else {
+				LOGGER.warn("Cannot remove unknown metric type {}. This indicates that the reporter " +
+					"does not support this metric type.", metric.getClass().getName());
+			}
+		}
+	}
+
+	@Override
+	public void open(MetricConfig config) {
+		client = new DatadogHttpClient(config.getString(API_KEY, null));
+		LOGGER.info("Configured DatadogHttpReporter");
+
+		configTags = getTagsFromConfig(config.getString(TAGS, null));
+	}
+
+	@Override
+	public void close() {
+		LOGGER.info("Shut down DatadogHttpReporter");
+	}
+
+	@Override
+	public void report() {
+		try {
+			DatadogHttpRequest request = new DatadogHttpRequest(this);
+
+			for(DGauge g : gauges.values()) {
+				try {
+					// Will throw exception if the Gauge is not of Number type
+					// Flink uses Gauge to store many types other than Number
+					g.getMetricValue();
+					request.addGauge(g);
+				} catch (Exception e) {
+					// ignore if the Gauge is not of Number type
+				}
+			}
+
+			for(DCounter c : counters.values()) {
```

Comment:
I'll fix them. 

But, is it enforced programmatically? I didn't see any checkstyle errors